datafusion/expr/src/expr.rs

```diff
@@ -1554,13 +1554,13 @@ impl Expr {
     /// Returns true if there are any column references in this Expr
     pub fn any_column_refs(&self) -> bool {
         self.exists(|expr| Ok(matches!(expr, Expr::Column(_))))
-            .unwrap()
+            .expect("exists closure is infallible")
```

Comment:
It's fine, but would it be possible to make `.exists` not return `Result` at all when closure doesn't return it?

would doing `Try` same way as `iter.try_fold` does work here too?

Reply:
I suppose we could rename `exists` to `try_exists` and then switch `exists` to not be fallable 🤔

Reply:
i tried to model this in a generic way with something like

````
F: Fn -> R
R: MaybeFallible<bool>
```

```
trait MaybeFallible<T> {
 fn is_falure(&self) -> bool;
 fn as_value(&self) -> Option<&T>;
```

It's sufficient to model the `exists` function signature generically, but i couldn't get exist and apply both working (perhaps didn't try long enough)